sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -665,15 +665,7 @@ case class AlterTableSetLocationCommand(
         catalog.alterPartitions(tableName, Seq(newPart))
       case None =>
         // No partition spec is specified, so we set the location for the table itself
-        val newTable =
-          if (DDLUtils.isDatasourceTable(table)) {
-            table.withNewStorage(
-              locationUri = Some(location),
-              properties = table.storage.properties ++ Map("path" -> location))
-          } else {
-            table.withNewStorage(locationUri = Some(location))
-          }
-        catalog.alterTable(newTable)
+        catalog.alterTable(table.withNewStorage(locationUri = Some(location)))
```

Comment:
It is legal to change the location for Hive managed tables. Why issuing an exception here?